x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/api/update.ts

```diff
@@ -44,6 +44,10 @@ export const registerSiemRuleMigrationsUpdateRoute = (
           const { migration_id: migrationId } = req.params;
           const rulesToUpdate = req.body;
 
+          const [firstRuleToUpdate] = rulesToUpdate;
+          if (!firstRuleToUpdate) {
+            return res.noContent();
+          }
```

Comment:
Consider just checking the array length
```suggestion
          if (rulesToUpdate.length === 0) {
            return res.noContent();
          }
```